modules/solutions/twitter-analytics/siddhi-extensions/siddhi-execution-tokenizer/component/src/main/java/org/wso2/extension/siddhi/execution/tokenizer/TweetTextTokenizer.java

```diff
@@ -0,0 +1,183 @@
+import org.wso2.siddhi.core.event.stream.StreamEventCloner;
+import org.wso2.siddhi.core.event.stream.populater.ComplexEventPopulater;
+import org.wso2.siddhi.core.exception.SiddhiAppCreationException;
+import org.wso2.siddhi.core.executor.ExpressionExecutor;
+import org.wso2.siddhi.core.query.processor.Processor;
+import org.wso2.siddhi.core.query.processor.stream.StreamProcessor;
+import org.wso2.siddhi.core.util.config.ConfigReader;
+import org.wso2.siddhi.query.api.definition.AbstractDefinition;
+import org.wso2.siddhi.query.api.definition.Attribute;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * splits a string into words
+ */
+
+@Extension(
+        name = "tokenize",
+        namespace = "text",
+        description = "This splits a string into words",
+        parameters = {
+                @Parameter(name = "text",
+                        description = "The input text which should be split.",
+                        type = {DataType.STRING}),
+        },
+        examples = @Example(
+                syntax = "define stream inputStream (text string);\n" +
+                        "@info(name = 'query1')\n" +
+                        "from inputStream#text:tokenize(text)\n" +
+                        "select text\n" +
+                        "insert into outputStream;",
+                description = "This query performs tokenization for the given string.")
+)
+
+public class TweetTextTokenizer extends StreamProcessor {
+    private static final Logger log = Logger.getLogger(TweetTextTokenizer.class);
+
+    @Override
+    protected void process(ComplexEventChunk<StreamEvent> streamEventChunk, Processor nextProcessor,
+                           StreamEventCloner streamEventCloner, ComplexEventPopulater complexEventPopulater) {
+        //Urls
+        String urlPattern = "(https?|ftp|file)://[-a-zA-Z0-9+&@#/%?=~_|!:,.;]*[-a-zA-Z0-9+&@#/%=~_|]";
+        // Punctuation chars
+        String punctChars = "[\\s+'“”‘’\\\".?!,:;&]";
+        String brackets = "[" + "<>«»{}\\(\\)\\[\\]" + "]";
+        String decorations = "[♫♪]+";
+        // Numeric
+        String timeLike = "\\d+:\\d+";
+        String numNum = "\\d+\\.\\d+";
+        Pattern pattern = Pattern.compile(punctChars + "|" + brackets + "|" + timeLike + "|" + numNum
+                + "|" + decorations);
+        while (streamEventChunk.hasNext()) {
+            StreamEvent streamEvent = streamEventChunk.next();
+            String event = (String) attributeExpressionExecutors[0].execute(streamEvent);
+            event = removeEmojis(event);
+            event = event.replaceAll(urlPattern, "").replaceAll("@(.*)", "").
```

Comment:
Please optimize this regex. Since you are only replacing with "", use a single regex to match all, and replace it once if possible.